stetho/src/main/java/com/facebook/stetho/common/android/FragmentCompatFramework.java

```diff
@@ -0,0 +1,115 @@
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.common.android;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.Fragment;
+import android.app.FragmentManager;
+import android.content.res.Resources;
+import android.os.Build;
+import android.view.View;
+
+import javax.annotation.Nullable;
+
+@TargetApi(Build.VERSION_CODES.HONEYCOMB)
+class FragmentCompatFramework
+    extends FragmentCompat<Fragment, FragmentManager, Activity> {
+  private static final FragmentAccessorFrameworkHoneycomb sFragmentAccessor;
+  private static final FragmentManagerAccessorViaReflection<FragmentManager, Fragment>
+      sFragmentManagerAccessor =
+          new FragmentManagerAccessorViaReflection<>(FragmentManager.class);
+  private static final FragmentActivityAccessorFramework sFragmentActivityAccessor =
+      new FragmentActivityAccessorFramework();
+
+  static {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+      sFragmentAccessor = new FragmentAccessorFrameworkJellyBean();
+    } else {
+      sFragmentAccessor = new FragmentAccessorFrameworkHoneycomb();
+    }
+  }
+
+  @Override
+  public Class<Fragment> getFragmentClass() {
+    return Fragment.class;
+  }
+
+  @Override
+  public FragmentAccessorFrameworkHoneycomb forFragment() {
+    return sFragmentAccessor;
+  }
+
+  @Override
+  public FragmentManagerAccessorViaReflection<FragmentManager, Fragment> forFragmentManager() {
+    return sFragmentManagerAccessor;
+  }
+
+  @Override
+  public FragmentActivityAccessorFramework forFragmentActivity() {
+    return sFragmentActivityAccessor;
+  }
+
+  private static class FragmentAccessorFrameworkHoneycomb
+      implements FragmentAccessor<Fragment, FragmentManager> {
+    @Nullable
+    @Override
+    public FragmentManager getFragmentManager(Fragment fragment) {
+      return fragment.getFragmentManager();
+    }
+
+    @Override
+    public Resources getResources(Fragment fragment) {
+      return fragment.getResources();
+    }
+
+    @Override
+    public int getId(Fragment fragment) {
+      return fragment.getId();
+    }
+
+    @Nullable
+    @Override
+    public String getTag(Fragment fragment) {
+      return fragment.getTag();
+    }
+
+    @Nullable
+    @Override
+    public View getView(Fragment fragment) {
+      return fragment.getView();
+    }
+
+    @Nullable
+    @Override
+    public FragmentManager peekChildFragmentManager(Fragment fragment) {
+      return null;
+    }
+  }
+
+  @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
+  private static class FragmentAccessorFrameworkJellyBean
+      extends FragmentAccessorFrameworkHoneycomb {
+    @Nullable
+    @Override
+    public FragmentManager peekChildFragmentManager(Fragment fragment) {
+      return fragment.getChildFragmentManager();
```

Comment:
This peek___() method is calling get___() ? In the old code, the point of implementing a peek___() method via reflection was to avoid the lazy-instantiation consequence of calling get___(). That said, now that I think about it, I'm not completely sure if this is really worth the optimization since there aren't going to be THAT many Fragments, right? If so, we might rename our peek___() to get___(). Not sure what the overall impact to application performance would be, though, since these FragmentManagers will still be there if Stetho is disconnected.